lasagne/updates.py

```diff
@@ -510,6 +512,52 @@ def adadelta(loss_or_grads, params, learning_rate=1.0, rho=0.95, epsilon=1e-6):
     return updates
 
 
+def adam(loss_or_grads, all_params, learning_rate=0.001, beta1=0.9,
+         beta2=0.999, epsilon=1e-8):
+    """Adam updates [1]_
+
+    References
+    ----------
+    [1] Diederik Kingma, Jimmy Ba,
+    Adam: A Method for Stochastic Optimization*,
+    http://arxiv.org/abs/1412.6980
+
+    Notes
+    ----------
+    The implementation does not include lambda as it is only used to
+    prove convergence of the algorithm.
+
+    Parameters
+    ----------
```

Comment:
Could you also add `loss_or_grads` and `all_params` to the parameter list in the docs, for completeness' sake? I guess they can be copied from another function.